### PR TITLE
feat(fury-adapter-swagger): add `termsOfService` support

### PR DESCRIPTION
**Pull request by marcofriso**
Closes APIARY-2822

### CODE REVIEW

File: packages/fury-adapter-swagger/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,19 @@
 # Fury Swagger Parser Changelog
 
+## 0.28.3 (2020-02-17)
```
Comment (kylef) [inline packages/fury-adapter-swagger/CHANGELOG.md]:
```suggestion
## Master
```

As this isn’t released yet; we leave the heading to “Master” (representing the master branch) and gets the number assigned when we make a release.

File: packages/fury-adapter-swagger/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+
+### Enhancements
+
+- Adds support for the `termsOfService` schema extension within the info section. 
+
+  For example, given following array schema:
```
Comment (kylef) [inline packages/fury-adapter-swagger/CHANGELOG.md]: Array schema? I don’t think this is related to the below example.

File: packages/fury-adapter-swagger/lib/parser.js [hunk 3]
```diff
@@ -325,6 +325,22 @@ class Parser {
...
+        if (this.swagger.info.termsOfService) {
+          this.withPath('termsOfService', () => {
+            const { Link } = this.namespace.elements;
+            const link = new Link();
+
+            link.relation = 'terms-of-service';
+            link.href = this.swagger.info.termsOfService;
+
+            if (this.generateSourceMap) {
+              this.createSourceMap(this.api.meta.get('termsOfService'), this.path);
```
Comment (kylef) [inline packages/fury-adapter-swagger/lib/parser.js]:
```suggestion
              this.createSourceMap(link, this.path);
```

We haven’t pushed the link into `this.api` yet so the source map wouldn’t end up in the right place. I think some of the other branches above such as for `title` might be broken. I’ll confirm when I’m back on Monday and advise how we can write test case for the source maps.

File: packages/fury-adapter-swagger/lib/parser.js [hunk 4]
```diff
@@ -325,6 +325,22 @@ class Parser {
...
+            }
+
+            return this.api.push(link);
```
Comment (kylef) [inline packages/fury-adapter-swagger/lib/parser.js]: The `link` element should go into the API Categories “links” meta property.

I think something like the following would work:

```suggestion
            return this.api.links.push(link);
```

File: packages/fury-adapter-swagger/test/fixtures/info-termsOfService.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          "content": "1.0.1"
+        }
+      },
+      "content": [
+        {
+          "element": "link",
+          "attributes": {
+            "relation": {
+              "element": "string",
+              "content": "terms-of-service"
```
Comment (tjanc) [inline packages/fury-adapter-swagger/test/fixtures/info-termsOfService.json]: Terms of service don't convey transactional semantics of an API. Wouldn't it better to add add it as a meta entry, alongside `title`?

Also, perhaps we should standardize this feature in https://github.com/apiaryio/api-elements before we merge an implementation.

File: packages/fury-adapter-swagger/CHANGELOG.md (hunk 1 above)
Comment (marcofriso) [reply packages/fury-adapter-swagger/CHANGELOG.md]: going to change

File: packages/fury-adapter-swagger/CHANGELOG.md (hunk 2 above)
Comment (marcofriso) [reply packages/fury-adapter-swagger/CHANGELOG.md]: indeed, going to change

File: packages/fury-adapter-swagger/lib/parser.js (hunk 3 above)
Comment (marcofriso) [reply packages/fury-adapter-swagger/lib/parser.js]: going to change, mind that if I comment out the whole 3 lines

```js
// if (this.generateSourceMap) {
//   this.createSourceMap(link, this.path);
// }
```

the test is still passing

File: packages/fury-adapter-swagger/lib/parser.js (hunk 4 above)
Comment (marcofriso) [reply packages/fury-adapter-swagger/lib/parser.js]: going to change

File: packages/fury-adapter-swagger/test/fixtures/info-termsOfService.json (hunk 5 above)
Comment (marcofriso) [reply packages/fury-adapter-swagger/test/fixtures/info-termsOfService.json]: with the next commit it will be under `link`, see if it is ok for you

Comment (tjanc) [reply packages/fury-adapter-swagger/test/fixtures/info-termsOfService.json]: Perfect.

File: packages/fury-adapter-swagger/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,20 @@
 # Fury Swagger Parser Changelog
 
+## Master
+
+### Enhancements
+
+- Adds support for the `info.termsOfService` schema extension.
```
Comment (kylef) [inline packages/fury-adapter-swagger/CHANGELOG.md]:
```suggestion
- Adds support for `info.termsOfService`. 
```

File: packages/fury-adapter-swagger/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,20 @@
 # Fury Swagger Parser Changelog
 
+## Master
+
+### Enhancements
+
+- Adds support for the `info.termsOfService` schema extension. 
+  For example the following schema:
```
Comment (kylef) [inline packages/fury-adapter-swagger/CHANGELOG.md]:
```suggestion
  For example the following document:
```

File: packages/fury-adapter-swagger/lib/parser.js (hunk 3 above)
Comment (kylef) [reply packages/fury-adapter-swagger/lib/parser.js]: I've created https://github.com/apiaryio/api-elements.js/pull/412 which we can rebase this PR onto once merged. Then you can have a test fixture named `info-termsOfService.sourcemap.json` which is similiar to `info-termsOfService.json` except also contains source maps.

File: packages/fury-adapter-swagger/CHANGELOG.md (hunk 6 above)
Comment (marcofriso) [reply packages/fury-adapter-swagger/CHANGELOG.md]: change on next commit

File: packages/fury-adapter-swagger/CHANGELOG.md (hunk 7 above)
Comment (marcofriso) [reply packages/fury-adapter-swagger/CHANGELOG.md]: change on next commit

**Comment by kylef** [comment]
@marcofriso can you rebase this PR onto master and add a next test fixture for the source maps. This is similiar to creating the initial test fixture:

```shell
$ echo '{}' > test/fixtures/info-termsOfService.sourcemap.json
$ env GENERATE=1 npm test
```

**Pull request merged**